### PR TITLE
Enhancement: update gitignore and remove generated frontend files

### PR DESCRIPTION
**Pull request by MustafaJafar**
# Changelog Description

Update gitignore list and remove generated frontend files.

## Additional Info
Commits in this PR brought from https://github.com/ynput/example-studio-addon/pull/1 as they seemed irrelevant.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,168 +1,22 @@
...
+frontend/yarn.lock
+frontend/yarn-error.log
+frontend/.yarn
+frontend/.pnp.cjs
+frontend/.pnp.loader.mjs
 
-node_modules/
\ No newline at end of file
+# Ignore genertated client version file
+client/*/version.py
```
Comment (iLLiCiTiT) [inline .gitignore]: Why? I don't want to create the file to be able to run from code...

File: .gitignore [hunk 2]
```diff
@@ -1,168 +1,22 @@
-# Ignore created packages
-/package
-
-/.idea
```
Comment (iLLiCiTiT) [inline .gitignore]: Why?

File: .gitignore [hunk 3]
```diff
@@ -1,168 +1,22 @@
...
-
-# Byte-compiled / optimized / DLL files
-__pycache__/
-*.py[cod]
-*$py.class
```
Comment (iLLiCiTiT) [inline .gitignore]: Why?

Comment (iLLiCiTiT) [reply .gitignore]: This is default gitignore for python projects
```
# Byte-compiled / optimized / DLL files
__pycache__/
*.py[cod]
*$py.class

# C extensions
*.so
```

And + the distribution

File: .gitignore (hunk 2 above)
Comment (iLLiCiTiT) [reply .gitignore]: This is for developers with PyCharm.

Comment (MustafaJafar) [reply .gitignore]: I have reverted that deletion. 
https://github.com/ynput/example-studio-addon/pull/3/commits/7c00197b79cfc43dbee52e9922f0aa486ec2a6bd

File: .gitignore (hunk 3 above)
Comment (MustafaJafar) [reply .gitignore]: I have reverted that deletion. 
https://github.com/ynput/example-studio-addon/pull/3/commits/7c00197b79cfc43dbee52e9922f0aa486ec2a6bd

File: .gitignore [hunk 4]
```diff
@@ -164,5 +164,12 @@ cython_debug/
...
\ No newline at end of file
+# Ignore generated frontend files
+node_modules
+dist
+frontend/yarn.lock
+frontend/yarn-error.log
+frontend/.yarn
+frontend/package-lock.json
+frontend/.pnp.cjs
+frontend/.pnp.loader.mjs
```
Comment (iLLiCiTiT) [inline .gitignore]:
```suggestion
frontend/.pnp.loader.mjs

```

File: .gitignore (hunk 3 above)
Comment (iLLiCiTiT) [reply .gitignore]: Aah, github not showing me 😥

**Review by iLLiCiTiT — APPROVED** [review]

**Pull request merged**